arch/cortex-m0/src/lib.rs

```diff
@@ -32,26 +32,22 @@ EXC_RETURN_PSP:
 #[inline(never)]
 /// r0 is top of user stack, r1 Process GOT
 pub unsafe extern "C" fn switch_to_user(mut user_stack: *const u8,
-                                        process_got: *const u8,
                                         process_regs: &mut [usize; 8])
                                         -> *mut u8 {
     asm!("
     /* Load non-hardware-stacked registers from Process stack */
-    ldmia $3!, {r4-r7}
+    ldmia $2!, {r4-r7}
     mov r11, r7
     mov r10, r6
     mov r9,  r5
     mov r8,  r4
-    ldmia $3!, {r4-r7}
-    subs $3, 32
+    ldmia $2!, {r4-r7}
+    subs $2, 32
 
     /* Load bottom of stack into Process Stack Pointer */
     msr psp, $0
 
-    /* Set PIC base pointer to the Process GOT */
-    mov r9, $2
-
-    mov r0, $3
+    mov r0, $2
```

Comment:
I had to re-think through this again and I think this could benefit from some comment cleanup, could you

  - move the `subs $2, 32` down to join this line
  - add a comment for the two instruction like `/* Reset pointer to kernel process register storage and save in r0 */`
  - Update the comment a few lines below to reflect where `r0` value actually comes from (the `(see to_kernel)` comment is wrong)